src/ls_server.rs

```diff
@@ -479,11 +485,13 @@ impl Output for StdioOutput {
 
 pub fn run_server(analysis: Arc<AnalysisHost>, vfs: Arc<Vfs>, build_queue: Arc<BuildQueue>) {
     let logger = Arc::new(Logger::new());
+    logger.log(&format!("\nLanguage Server Starting up\n"));
     let service = LsService::new(analysis,
                                  vfs,
                                  build_queue,
                                  Box::new(StdioMsgReader { logger: logger.clone() }),
                                  Box::new(StdioOutput { logger: logger.clone() } ),
-                                 logger);
+                                 logger.clone());
     LsService::run(service);
+    logger.log(&format!("\nServer shutting down.\n"));
```

Comment:
There's a number of output messages which don't end in a trailing newline, this is here to make sure this is on it's own line, for readability.
Should I try and tackle the missing newlines elsewhere instead?